From da14e436bd5e1f70f7f54eee2af48a2f8fa699d4 Mon Sep 17 00:00:00 2001
From: Philippe Renon <philippe_renon@yahoo.fr>
Date: Tue, 6 Feb 2018 16:16:32 +0100
Subject: [PATCH 1/2] setup link libraries: change order of target link
 libraries

make sure the external libraries are added last
this addresses a linking error when building cache_rocksdb where
rocksdb dependencies were listed before librocksdb itself
---
 CMakeModules/OsgEarthMacroUtils.cmake | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/CMakeModules/OsgEarthMacroUtils.cmake b/CMakeModules/OsgEarthMacroUtils.cmake
index e1bd5d488..d3a89221f 100644
--- a/CMakeModules/OsgEarthMacroUtils.cmake
+++ b/CMakeModules/OsgEarthMacroUtils.cmake
@@ -169,12 +169,13 @@ MACRO(SETUP_LINK_LIBRARIES)
 #    ENDFOREACH(LINKLIB)
     LINK_INTERNAL(${TARGET_TARGETNAME} ${TARGET_LIBRARIES})
 
+    IF(TARGET_LIBRARIES_VARS)
+            LINK_WITH_VARIABLES(${TARGET_TARGETNAME} ${TARGET_LIBRARIES_VARS})
+    ENDIF(TARGET_LIBRARIES_VARS)
+
     FOREACH(LINKLIB ${TARGET_EXTERNAL_LIBRARIES})
             TARGET_LINK_LIBRARIES(${TARGET_TARGETNAME} ${LINKLIB})
     ENDFOREACH(LINKLIB)
-        IF(TARGET_LIBRARIES_VARS)
-            LINK_WITH_VARIABLES(${TARGET_TARGETNAME} ${TARGET_LIBRARIES_VARS})
-        ENDIF(TARGET_LIBRARIES_VARS)
 ENDMACRO(SETUP_LINK_LIBRARIES)
 
 ############################################################################################
-- 
2.16.1

